arch/x86/include/mmustructs.h

```diff
@@ -129,17 +129,26 @@ struct mmu_region {
  * region_size has to be provided in bytes
  * for read write access = MMU_ENTRY_READ/MMU_ENTRY_WRITE
  * for supervisor/user mode access = MMU_ENTRY_SUPERVISOR/MMU_ENTRY_USER
+ *
+ * Preprocessor indirection layers used to ensure __COUNTER__ is expanded
+ * properly.
  */
 
-#define MMU_BOOT_REGION(addr, region_size, permission_flags)		\
-	static struct mmu_region region_##addr				\
+#define __MMU_BOOT_REGION(id, addr, region_size, permission_flags)	\
+	static struct mmu_region region_##id				\
 	__attribute__((__section__(".mmulist"), used))  =		\
 	{								\
-		.address = addr,					\
-		.size = region_size,					\
+		.address = (u32_t)(addr),				\
+		.size = (u32_t)(region_size),				\
```

Comment:
Typesafety whine: Seems like if the user is passing an incompatible type to the macro, that's an error we want flagged and not suppressed.   the region_size argument should 100% absolutely be a size_t value already or it's a bug.  It seems like maybe you want to define the address field as a void* and not an integer so you can typecheck the macro, but maybe that's an API we don't want to modify.